packages/core/src/utils/components.ts

```diff
@@ -7,18 +12,34 @@ const structureComponentIds = new Set([
   CONTENTFUL_COMPONENTS.singleColumn.id,
 ]);
 
+const patternTypes = new Set([
+  ASSEMBLY_NODE_TYPE,
+  ASSEMBLY_DEFAULT_CATEGORY,
+  ASSEMBLY_BLOCK_NODE_TYPE,
+]);
+
+type ComponentAllowOnRoot = {
+  type?: string;
+  category?: string;
+  componentId?: string;
+};
+
 const allContentfulComponentIds = new Set(
   Object.values(CONTENTFUL_COMPONENTS).map((component) => component.id),
 );
 
+export const isPatternComponent = (type?: string) => patternTypes.has(type ?? '');
+
 export const isContentfulStructureComponent = (componentId?: string) =>
   structureComponentIds.has(componentId ?? '');
 
 export const isContentfulComponent = (componentId?: string) =>
   allContentfulComponentIds.has(componentId ?? '');
 
-export const isComponentAllowedOnRoot = (componentId?: string) =>
-  isContentfulStructureComponent(componentId) || componentId === CONTENTFUL_COMPONENTS.divider.id;
+export const isComponentAllowedOnRoot = ({ type, category, componentId }: ComponentAllowOnRoot) =>
+  isPatternComponent(type || category) ||
```

Comment:
I think separating `type` and `category` into two different params makes this code cleaner and easier to follow!

However, I still think the `patternTypes` Set including a category seems confusing and I would suggest removing it.  Instead, since there's only one category, you can check that as a separate condition, like this:

```suggestion
export const isComponentAllowedOnRoot = ({ type, category, componentId }: ComponentAllowOnRoot) =>
  isPatternComponent(type) ||
  category === ASSEMBLY_DEFAULT_CATEGORY ||
```